content/ytt/docs/latest/lang-ref-ytt-schema.md

```diff
@@ -0,0 +1,248 @@
+---
+title: Schema module
+---
+
+## Overview
+
+`ytt`'s Schema feature provides a way to verify YAML structures and values with the help of annotations.
+Schema currently allows you to provide default data values, and verify that each data value file provided is compatible with the schema.
+
+The schema definition spells out the set of nodes that must make up that kind of YAML document. Documents that are type checked and validated against the schema are guaranteed to be proper.
+
+For example, in the case of Data Values, the schema guarantees to templates that all values exist and are of the correct type. The aleviates templates from doing existence and type checks, themselves.
+
+There are two features to verify a document is compatible with a schema:
+- type checking — comparing the shape of a YAML structure against its schema’s types.
+- validation — determining whether the values contained in a YAML structure are acceptable.
+---
+## Supported Types
+
+While YAML provides for an extendable range of types, the `ytt` Schema supports a specific set.
+
+### Scalar Types
+
+- `bool` — `true`, `false` (and when not strict, `yes`, `no`, `Y`, `N`, etc.)
+- `float` — e.g. `0.4`
+- `int` — e.g. `42`
+- `null` — `null`, `~`, and when value is omitted.
+- `string` — e.g. `""`, `"ConfigMap"`, `"0xbeadcafe"`
+
+
+### Algebraic Types
+
+- `any` — any valid YAML value is permitted.
+
+### Collection Types
+
+A collection type is the pair of itself (either `map` or `array`) and its contained type.
+
+#### Array type
+
+`array<type>` — a YAML sequence, each item must be of the specified `type`.
+
+Examples:
+- `array<string>` — a sequence of strings (and only strings)
+- `array<map<...>>` — a sequence containing only map instances of the same shape.
+
+#### Map type
+
+`map <keyA:typeA, keyB:typeB, ...>` — a YAML mapping that must contain exactly one item of each given key, whose value must be of the type specified.
+
```

Comment:
Let's simplify this:
- introduce as few concepts as possible (e.g. remove "Algebraic Types" as an idea and explain `any` a special type?
- I don't think anything is lost if we merely refer to "Arrays" and "Maps"... we've been somewhat consistent in referring to their direct children as "array item" and "map item"
- the angle-bracket notation isn't showing up in any interface the user sees and might be confusing to introduce it here (as if it's a notation they should be aware of).  Instead, perhaps give concrete examples?

Reply:
I modified this section to remove algebraic types, and kept maps and arrays in their own sections but provided examples. I am torn between putting maps and arrays with the other types, or keeping them on their own.